conformance/utils/suite/experimental_suite.go

```diff
@@ -0,0 +1,289 @@
+limitations under the License.
+*/
+
+package suite
+
+import (
+	"fmt"
+	"sync"
+	"testing"
+	"time"
+
+	"k8s.io/apimachinery/pkg/util/sets"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"sigs.k8s.io/gateway-api/conformance"
+	confv1a1 "sigs.k8s.io/gateway-api/conformance/apis/v1alpha1"
+	"sigs.k8s.io/gateway-api/conformance/utils/config"
+	"sigs.k8s.io/gateway-api/conformance/utils/kubernetes"
+	"sigs.k8s.io/gateway-api/conformance/utils/roundtripper"
+)
+
+// -----------------------------------------------------------------------------
+// Conformance Test Suite - Public Types
+// -----------------------------------------------------------------------------
+
+// ConformanceTestSuite defines the test suite used to run Gateway API
+// conformance tests.
+// This is experimental for now and can be used as an alternative to the
+// ConformanceTestSuite. Once this won't be experimental any longer,
+// the two of them will be merged.
+type ExperimentalConformanceTestSuite struct {
+	ConformanceTestSuite
+
+	// implementation contains the details of the implementation, such as
+	// organization, project, etc.
+	implementation confv1a1.Implementation
+
+	// conformanceProfiles is a compiled list of profiles to check
+	// conformance against.
+	conformanceProfiles sets.Set[ConformanceProfileName]
+
+	// running indicates whether the test suite is currently running
+	running bool
+
+	// results stores the pass or fail results of each test that was run by
+	// the test suite, organized by the tests unique name.
+	results map[string]testResult
+
+	// extendedSupportedFeatures is a compiled list of named features that were
+	// marked as supported, and is used for reporting the test results.
+	extendedSupportedFeatures map[ConformanceProfileName]sets.Set[SupportedFeature]
+
+	// extendedUnsupportedFeatures is a compiled list of named features that were
+	// marked as not supported, and is used for reporting the test results.
+	extendedUnsupportedFeatures map[ConformanceProfileName]sets.Set[SupportedFeature]
+
+	// lock is a mutex to help ensure thread safety of the test suite object.
+	lock sync.RWMutex
+}
+
+// Options can be used to initialize a ConformanceTestSuite.
+type ExperimentalConformanceOptions struct {
+	Options
+
+	Implementation      confv1a1.Implementation
+	ConformanceProfiles sets.Set[ConformanceProfileName]
+}
+
+// NewExperimentalConformanceTestSuite returns a new ExperimentalConformanceTestSuite.
+// NewExperimentalConformanceTestSuite is a helper to use for creating the ExperimentalConformanceTestSuite.
+// The implementations willing to run such a suite to obtain the profile badge need to create the
+// ExperimentalConformanceTestSuite by means of this helper and pass an instance of
+// ExperimentalConformanceOptions containing all the implementation details contained in the standard
+// ConformanceSuiteOptions plus the implementation details and the profile against which to verify
+// conformance.
```

Comment:
Tiny non-blocking nit: This feels more verbose than it needs to be and mentions concepts that we haven't actually defined yet (a profile badge). I think it would be fine if this just said `NewExperimentalConformanceTestSuite is a helper to use for creating a new ExperimentalConformanceTestSuite` or something similar.

Reply:
I changed the comment as you suggested :)